osgp/protocol-adapter-oslp/web-device-simulator/src/main/java/org/opensmartgridplatform/webdevicesimulator/service/OslpChannelHandler.java

```diff
@@ -569,307 +778,65 @@ private void sleep(final Long sleepTime) {
-                            .addRelayData(Oslp.RelayData.newBuilder()
-                                    .setIndex(ByteString.copyFrom(new byte[] { 4 }))
-                                    .setTotalLightingMinutes(burningMinutes - randomCumulativeMinutes)))
-                    .build();
+            response = createSetTransitionResponse();
+        } else if (request.hasConfirmRegisterDeviceRequest()) {
+            response = createConfirmRegisterDeviceResponse(request.getConfirmRegisterDeviceRequest().getRandomDevice(),
+                    request.getConfirmRegisterDeviceRequest().getRandomPlatform());
+        } else {
+            // Handle errors by logging
+            LOGGER.error("Did not expect request, ignoring: {}", request);
+        }
 
-            powerUsageDataList.add(powerUsageData);
-            pageStartTime = pageStartTime.minusMinutes(intervalMinutes);
+        return response;
+    }
 
-            burningMinutes -= CUMULATIVE_BURNING_MINUTES;
+    private void sendDelayedDeviceRegistration(final Device device) {
+        if (device == null) {
+            return;
         }
 
-        return createUsageMessage(currentPageNumber, itemsPerPage, numberOfPages, powerUsageDataList);
-    }
+        final String deviceIdentification = device.getDeviceIdentification();
+        if (StringUtils.isEmpty(deviceIdentification)) {
+            return;
+        }
 
-    private static DateTime correctUsageUntilDate(final DateTime dateTimeUntil, final HistoryTermType termType) {
-        final DateTime now = new DateTime();
-        if (dateTimeUntil.isAfter(now)) {
-            if (termType == HistoryTermType.Short) {
-                return now.hourOfDay().roundCeilingCopy();
-            } else {
-                return now.withZone(localTimeZone).dayOfWeek().roundCeilingCopy().withZone(DateTimeZone.UTC);
+        new Timer().schedule(new TimerTask() {
+
+            @Override
+            public void run() {
+                try {
+                    LOGGER.info("Sending DeviceRegistrationRequest for device: {}", deviceIdentification);
+                    final DeviceMessageStatus deviceMessageStatus = OslpChannelHandler.this.registerDevice
+                            .sendRegisterDeviceCommand(device.getId(), true);
+                    if (DeviceMessageStatus.OK.equals(deviceMessageStatus)) {
+                        LOGGER.info("Sending ConfirmDeviceRegistrationRequest for device: {}", deviceIdentification);
+                        OslpChannelHandler.this.registerDevice.sendConfirmDeviceRegistrationCommand(device.getId());
+                    }
+                } catch (final Exception e) {
+                    LOGGER.error("Caught exception during sendDelayedDeviceRegistration() for device : "
+                            + deviceIdentification, e);
+                }
             }
-        }
 
-        return dateTimeUntil;
+        }, 2000);
```

Comment:
If the value `2000` makes sense in all kinds of scenarios it is probably good enough for now.
If we can already foresee that it would come in handy to set this to smaller or larger values for local cucumber tests, general test environment use and something like nightly tests for instance, it may be good to make it configurable, for example through the properties file.

Reply:
This is an existing delay, not updating or moving it to properties for now.